internal/pkg/agent/application/dispatcher/dispatcher.go

```diff
@@ -321,7 +321,7 @@ func reportNextScheduledUpgrade(input []fleetapi.Action, detailsSetter details.O
 
 	upgradeDetails := details.NewDetails(nextUpgrade.Version, details.StateScheduled, nextUpgrade.ID())
 	startTime, _ := nextUpgrade.StartTime()
-	upgradeDetails.Metadata.ScheduledAt = startTime
+	upgradeDetails.Metadata.ScheduledAt = &startTime
```

Comment:
I am not 100% sure this is correct, we probably need to test startTime for zero value (checked the `StartTime()` function)

Reply:
see 73a9031524ab62212f12eef2428c29ea620aa458

Reply:
It built before because two PRs independently succeeded in CI.  So they got merged into `main`.  But after that, the changes in them collided and we got this failure.

To prevent this in the future, we need to enable the *Require branches to be up to date before merging branch* protection.